modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -998,24 +1012,23 @@ else if (!deleteRelationshipObjectField) {
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectField.getObjectDefinitionId());
 
+		if (objectDefinition.isSystem() && objectField.isSystem() &&
+			!SystemObjectDefinitionManagementChecker.isInvokerBundleAllowed()) {
+
+			throw new ObjectFieldSystemException(
+				"Only allowed bundles can delete system object fields");
+		}
+
 		int customObjectFieldsCount =
 			objectFieldLocalService.getObjectFieldsCount(
 				objectField.getObjectDefinitionId(), false);
 
-		if (objectDefinition.isApproved() &&
-			!objectDefinition.isUnmodifiableSystemObject() &&
+		if (objectDefinition.isApproved() && !objectDefinition.isSystem() &&
```

Comment:
This condition verify the deletion of custom fields for custom object definition.

I'm thinking that we need to verify the deletion of fields when the object definition is system/modifiable; (add a condition)

What do you think?

Reply:
If this is too much work, we can add this in another PR.

Reply:
It'd be better if we add this changes in this PR.
I will push the changes here, but if we change our mind we can remove the commits related to this changes and add in another PR. :+1: